pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -942,7 +942,19 @@ func (a *ApplicationBackupController) backupVolumes(backup *stork_api.Applicatio
 					return fmt.Errorf("error getting backup status: %v", err)
 				}
 				for _, volInfo := range volumeInfos {
-					if volInfo.BackupID == "" {
+					if volInfo.Status == stork_api.ApplicationBackupStatusFailed {
+						errorMsg := fmt.Sprintf("Error backing up volume %v from namespace: %v : %v", volInfo.Volume, volInfo.Namespace, volInfo.Reason)
+						a.recorder.Event(backup,
+							v1.EventTypeWarning,
+							string(volInfo.Status),
+							errorMsg)
+
+						backup.Status.Stage = stork_api.ApplicationBackupStageFinal
+						backup.Status.FinishTimestamp = metav1.Now()
+						backup.Status.Status = stork_api.ApplicationBackupStatusFailed
+						backup.Status.Reason = errorMsg
+						break
```

Comment:
why do we need to break here and continue out side of for loop. You are setting StageFinal and status Failed. We should be return nil for reconciler to take care of failure case. 
And I think you need to incorporate partial success here.. This is a partial success for us. 
I feel instead of handling this through this bug, it is better partial success implementation implicitly takes care of it.

Reply:
@vsundarraj-px 
- the break here is to ensure we don't check any other volume status once one has failed
- the outer for loop is to check only portworx driver, also we continue outside the for loop so that post exec rules run properly
- This has been handled in partial success but since partial success will be implemented in 2.7.2, we have made this fix for 2.7.1
- if we return nil from here, the reconciler will keep looping in "Volume in progress" state

Reply:
@shkumari-px It is better to return instead of break. The execution of posterec rule in the case of failure happens in the DeleteTimestamp handling 
https://github.com/libopenstorage/stork/blob/f97b096d1235815924fb1dc23eed6e5700b22f2c/pkg/applicationmanager/controllers/applicationbackup.go#L285

Reply:
@vsundarraj-px https://github.com/libopenstorage/stork/blob/f97b096d1235815924fb1dc23eed6e5700b22f2c/pkg/applicationmanager/controllers/applicationbackup.go#L285 won't take care of running post exec rule since, post exec rule is being run here only when backup is in volume stage